ssas/service/public/api.go

```diff
@@ -249,7 +249,7 @@ func token(w http.ResponseWriter, r *http.Request) {
 		service.JSONError(w, http.StatusUnauthorized, http.StatusText(http.StatusUnauthorized), "invalid client id")
 		return
 	}
-	err = ValidateSecret(system, secret, w, r)
+	err = ValidateSecret(system, secret, r)
 	if err != nil {
 		ssas.Logger.Error("The client id and secret cannot be validated: ", err.Error())
```

Comment:
because we have removed the `service.JSONError(...)` in `ValidateSecret`, we should add `JsonError` to the line below the log statement here so we are still returning the error/http response.

Reply:
this looks good! One thing to consider is that we are changing the return value of this call from one of three options [`Error getting secret`, `The incoming client secret is invalid`, or `credentials expired`] to `Error getting secret`.  

I had a quick look through bcda-app and we are only returning the http status to our consumers and we aren't using the error message returned from `validateSecret` elsewhere, so this change should be fine but we should keep an eye out, just in case.